pkg/endpoint/bpf.go

```diff
@@ -1374,25 +1380,33 @@ func (e *Endpoint) syncPolicyMapWithDump() error {
 	return err
 }
 
+const (
+	// syncPolicyMapInterval is the interval for periodic full reconciliation of
+	// the policy map to catch unexpected discrepancies with agent and kernel state.
+	syncPolicyMapInterval = 15 * time.Minute
+)
+
 func (e *Endpoint) syncPolicyMapController() {
-	ctrlName := fmt.Sprintf("sync-policymap-%d", e.ID)
-	e.controllers.UpdateController(ctrlName,
-		controller.ControllerParams{
-			Group: syncPolicymapControllerGroup,
-			DoFunc: func(ctx context.Context) (reterr error) {
-				// Failure to lock is not an error, it means
-				// that the endpoint was disconnected and we
-				// should exit gracefully.
-				if err := e.lockAlive(); err != nil {
-					return controller.NewExitReason("Endpoint disappeared")
-				}
-				defer e.unlock()
-				return e.syncPolicyMapWithDump()
+	e.startSyncOnce.Do(func() {
```

Comment:
Opted to do this via `sync.Once` instead of changing where `syncPolicyMapController` is invoked from as there were multiple places allocating an endpoint (createEndpoint, parseEndpoint, ...) and no definite "start the endpoint" method. So seemed like keeping the original structure was safest.

Reply:
This kind of invocation pattern is what `pkg/trigger` provides, but what it misses is the retrying ability of `pkg/controller`. A bit unfortunate that there's no mechanism to have both, but `sync.Once` works for now.

Reply:
This made me think we should just add this ability to `pkg/controller`. I added a `CreateController` method that creates the controller if it doesn't exist and does nothing if it does. This way we don't need the `sync.Once` etc.
WDYT?